src/otx/core/data/dataset/instance_segmentation.py

```diff
@@ -0,0 +1,76 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Module for OTXInstanceSegDataset."""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import numpy as np
+import torch
+from datumaro import DatasetSubset, Image, Polygon
+from torchvision import tv_tensors
+
+from otx.core.data.entity.base import ImageInfo
+from otx.core.data.entity.instance_segmentation import InstanceSegBatchDataEntity, InstanceSegDataEntity
+from otx.core.utils.mask_util import polygon_to_bitmap
+
+from .base import OTXDataset, Transforms
+
+
+class OTXInstanceSegDataset(OTXDataset[InstanceSegDataEntity]):
+    """OTXDataset class for instance segmentation."""
+
+    def __init__(self, dm_subset: DatasetSubset, transforms: Transforms) -> None:
+        super().__init__(dm_subset, transforms)
+        self.poly2mask = 'polygon' not in dm_subset.get_annotated_type()
+
+    def _get_item_impl(self, index: int) -> InstanceSegDataEntity | None:
+        item = self.dm_subset.get(id=self.ids[index], subset=self.dm_subset.name)
+        img = item.media_as(Image)
+        img_data = self._get_img_data(img)
+        img_shape = img.size
+
+        gt_bboxes = np.zeros(shape=(0, 4), dtype=np.float32)
+        gt_labels = np.zeros(shape=(0, ), dtype=int)
+        gt_masks = np.zeros(shape=(0, *img_shape), dtype=bool)
+        gt_polygons = []
+        for annotation in item.annotations:
+            if isinstance(annotation, Polygon):
+                points = np.array(annotation.points).reshape(-1, 2).astype(np.int32)
+                x1, y1 = np.min(points, axis=0)
+                x2, y2 = np.max(points, axis=0)
```

Comment:
```suggestion
                bbox = np.array(annotation.get_bbox(), dtype=np.int32) # xywh
```

1. Please reuse `Datumaro`'s API to make it looking clean
2. Please `np.array(..., dtype=...)` rather than `np.array(...).astype(...)` because there is an overhead to convert the Numpy array's dtype after construction.